servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/amqp/ReactorDispatcher.java

```diff
@@ -54,105 +51,80 @@ private void initializeSelectable()
         this.reactor.update(schedulerSelectable);
     }
 
-    public void invoke(final DispatchHandler timerCallback) throws IOException
-    {
+    public void invoke(final DispatchHandler timerCallback) throws IOException {
         this.workQueue.offer(timerCallback);
         this.signalWorkQueue();
     }
 
-    public void invoke(final int delay, final DispatchHandler timerCallback) throws IOException
-    {
+    public void invoke(final int delay, final DispatchHandler timerCallback) throws IOException {
         this.workQueue.offer(new DelayHandler(this.reactor, delay, timerCallback));
         this.signalWorkQueue();
     }
 
-    private void signalWorkQueue() throws IOException
-    {
-        try
-        {
+    private void signalWorkQueue() throws IOException {
+        try {
             this.ioSignal.sink().write(ByteBuffer.allocate(1));
-        }
-        catch(ClosedChannelException ignorePipeClosedDuringReactorShutdown)
-        {
+        } catch (ClosedChannelException ignorePipeClosedDuringReactorShutdown) {
         }
     }
 
-    private final class DelayHandler extends BaseHandler
-    {
+    private static final class DelayHandler extends BaseHandler {
```

Comment:
[SpotBugs-P2] 
This class is an inner class, but does not use its embedded reference to the object which created it.  This reference makes the instances of the class larger, and may keep the reference to the creator object alive longer than necessary.  If possible, the class should be made static.